pkg/cgroups/systemd_linux.go

```diff
@@ -140,6 +140,10 @@ func resourcesToProps(res *configs.Resources) (map[string]uint64, map[string]str
 		uMap["CPUQuotaPerSecUSec"] = cpuQuotaPerSecUSec
 	}
 
+	if res.CpuShares != 0 {
+		uMap["CPUWeight"] = res.CpuShares
```

Comment:
I think we need to convert from CPU shares to CPU weight.  While they are the same conceptually, they have a different scale.

You can look at the crun function `convert_shares_to_weight` that uses a linear conversion, or the systemd conversion.